src/main/java/slimeknights/tconstruct/gadgets/item/slimesling/PurpleSlimeSlingItem.java

```diff
@@ -0,0 +1,51 @@
+package slimeknights.tconstruct.gadgets.item.slimesling;
+
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.player.ServerPlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.particles.ParticleTypes;
+import net.minecraft.util.SoundEvents;
+import net.minecraft.util.math.vector.Vector3d;
+import net.minecraft.world.World;
+import slimeknights.tconstruct.common.Sounds;
+import slimeknights.tconstruct.library.network.TinkerNetwork;
+import slimeknights.tconstruct.tools.common.network.EntityMovementChangePacket;
+
+public class PurpleSlimeSlingItem extends BaseSlimeSlingItem {
+
+  public PurpleSlimeSlingItem(Properties props) {
+    super(props);
+  }
+
+  /** Called when the player stops using an Item (stops holding the right mouse button). */
+  @Override
+  public void onPlayerStoppedUsing(ItemStack stack, World worldIn, LivingEntity entityLiving, int timeLeft) {
+    if (!(entityLiving instanceof PlayerEntity)) {
+      return;
+    }
+
+    PlayerEntity player = (PlayerEntity) entityLiving;
+    float f = getForce(stack, timeLeft);
+
+    Vector3d look = player.getLookVec();
+    double offX = look.x * f;
+    double offY = look.y * f + 1;
+    double offZ = look.z * f;
+
+    if (player.attemptTeleport(player.getPosX() + offX, player.getPosY() + offY, player.getPosZ() + offZ, true)) {
```

Comment:
This feels awkward when it cannot teleport, as the player does not move. If I am staring at a block, and teleporting would put me inside it, I'd expect to teleport right next to it.

Not sure of the best way to do this, I suppose we could start from the block position and iterate towards the player backwards along the look vector. Or maybe we could approximate the behavior by just selecting a second position to try via raytrace if the first does not move the player.

Also, this behaves pretty weirdly in water, we might want to write our own attempt teleport method that lets us ignore liquids.